app/code/Magento/Backend/Block/Widget/Grid/Column/Renderer/Action.php

```diff
@@ -144,7 +145,7 @@ protected function _transformActionData(&$action, &$actionCaption, \Magento\Fram
                     if (is_array($action['url']) && isset($action['field'])) {
                         $params = [$action['field'] => $this->_getValue($row)];
                         if (isset($action['url']['params'])) {
-                            $params = array_merge($action['url']['params'], $params);
+                            $params[] = $action['url']['params'];
```

Comment:
In my opinion - this line introduces the bug, as previously you had:
```php
$action['url']['params'] = ['a' => 'A', 'b' => 'B'];
$params = ['x' => 'X'];

$params = array_merge($action['url']['params'], $params);
```

resulted in having keys `a`, `b` and `x`.

The new approach results in:
```
'x' => 'X',
[
    'a' => 'A',
    'b' => 'B'
]
```

Reply:
Good catch! Could you create Pull request for that?